CMakeLists.txt

```diff
@@ -92,29 +92,32 @@ add_definitions(-DHAVE_FALLBACK)
 
 
 ## Functions and headers
-include (CheckFunctionExists)
-include (CheckIncludeFile)
-macro (check_funcs) # Similar to AC_CHECK_FUNCS of autotools
-  foreach (func_name ${ARGN})
-    string(TOUPPER ${func_name} definiton_to_add)
-    check_function_exists(${func_name} HAVE_${definiton_to_add})
-    if (${HAVE_${definiton_to_add}})
-      add_definitions(-DHAVE_${definiton_to_add})
-    endif ()
-  endforeach ()
-endmacro ()
-check_funcs(atexit mprotect sysconf getpagesize mmap isatty newlocale strtod_l)
-check_include_file(unistd.h HAVE_UNISTD_H)
-if (${HAVE_UNISTD_H})
-  add_definitions(-DHAVE_UNISTD_H)
-endif ()
-check_include_file(sys/mman.h HAVE_SYS_MMAN_H)
-if (${HAVE_SYS_MMAN_H})
-  add_definitions(-DHAVE_SYS_MMAN_H)
-endif ()
-check_include_file(xlocale.h HAVE_XLOCALE_H)
-if (${HAVE_XLOCALE_H})
-  add_definitions(-DHAVE_XLOCALE_H)
+if (NOT EMSCRIPTEN) # time consuming and not needed for Emscripten builds
+  include (CheckFunctionExists)
+  include (CheckIncludeFile)
+  macro (check_funcs) # Similar to AC_CHECK_FUNCS of autotools
+    foreach (func_name ${ARGN})
+      string(TOUPPER ${func_name} definiton_to_add)
+      CHECK_FUNCTION_EXISTS(${func_name} HAVE_${definiton_to_add})
+      if (${HAVE_${definiton_to_add}})
+        add_definitions(-DHAVE_${definiton_to_add})
+      endif ()
+    endforeach ()
+  endmacro ()
```

Comment:
The `include` and `macro` lines only add generic functions, so may be worthwhile to hoist them out from the `if (NOT EMSCRIPTEN)` path and restrict the Emscripten-specific changes to as few lines as possible - probably only over the `check_funcs` lines? That way there won't be any misassumptions that something e.g. in the `check_funcs` macro would be incompatible with Emscripten.

Also if this was done like the comment suggests only to speed up Emscripten CMake runs, then I think it might be desirable to remove this `if (NOT EMSCRIPTEN)` part altogether, and arrive to the same speed wins by pre-specifying all the `-DHAVE_UNISTD_H=1` etc. from the CMake command line. If I understand correctly, CMake only does these checks if the variables have not yet been checked, so pre-setting the results on the command line should have the same effect.

There is a chance that in the distant future the Emscripten environment changes and one of the headers, like `xlocale.h` might begin to exist, so there's also a chance that avoiding all this configuration would lead to incorrect results?

Reply:
Thanks for you review! Done.